xcp_d/utils/hcp2fmriprep.py

```diff
@@ -4,165 +4,310 @@
+
+    # Grab surface morphometry files
+    anat_dir_orig = in_dir
+    fsaverage_dir_orig = os.path.join(anat_dir_orig, "MNINonLinear/fsaverage_LR32k")
+
+    SURFACE_DICT = {
+        "R.midthickness": "hemi-R_desc-hcp_midthickness",
+        "L.midthickness": "hemi-L_desc-hcp_midthickness",
+        "R.inflated": "hemi-R_desc-hcp_inflated",
+        "L.inflated": "hemi-L_desc-hcp_inflated",
+        "R.very_inflated": "hemi-R_desc-hcp_vinflated",
+        "L.very_inflated": "hemi-L_desc-hcp_vinflated",
+        "R.pial": "hemi-R_pial",
+        "L.pial": "hemi-L_pial",
+        "R.white": "hemi-R_smoothwm",
+        "L.white": "hemi-L_smoothwm",
+    }
+    for in_str, out_str in SURFACE_DICT.items():
+        surf_orig = os.path.join(
+            fsaverage_dir_orig,
+            f"{sub_id}.{in_str}.32k_fs_LR.surf.gii",
+        )
+        surf_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_ent}_space-fsLR_den-32k_{out_str}.surf.gii",
+        )
+        copy_dictionary[surf_orig] = [surf_fmriprep]
 
-    # to fmriprep directory
-    if "sub" not in sub_id:
-        subid = "sub-" + sub_id
-    t1wim = anatdir + subid + "_desc-preproc_T1w.nii.gz"
-    t1seg = anatdir + subid + "_dseg.nii.gz"
-    t1ribbon = anatdir + subid + "_desc-ribbon_T1w.nii.gz"
-    t1brainm = anatdir + subid + "_desc-brain_mask.nii.gz"
-    regfile1 = anatdir + subid + "_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.txt"
-    regfile2 = anatdir + subid + "_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.txt"
-
-    lMid = anatdir + subid + "_hemi-L_midthickness.surf.gii"
-    rMid = anatdir + subid + "_hemi-R_midthickness.surf.gii"
-
-    lpial = anatdir + subid + "_hemi-L_pial.surf.gii"
-    rpial = anatdir + subid + "_hemi-R_pial.surf.gii"
-
-    lwhite = anatdir + subid + "_hemi-L_smoothwm.surf.gii"
-    rwhite = anatdir + subid + "_hemi-R_smoothwm.surf.gii"
-
-    linf = anatdir + subid + "_hemi-L_inflated.surf.gii"
-    rinf = anatdir + subid + "_hemi-R_inflated.surf.gii"
-    newanatfiles = [
-        t1wim,
-        t1seg,
-        t1ribbon,
-        t1brainm,
-        regfile1,
-        regfile2,
-        lMid,
-        rMid,
-        lpial,
-        rpial,
-        lwhite,
-        rwhite,
-        linf,
-        rinf,
+    print("finished collecting anat files")
+
+    # Collect functional files to copy
+    subject_task_folders = sorted(
+        glob.glob(os.path.join(anat_dir_orig, "MNINonLinear/Results", "*"))
+    )
+    subject_task_folders = [
+        task for task in subject_task_folders if task.endswith("RL") or task.endswith("LR")
     ]
+    for subject_task_folder in subject_task_folders:
+        # NOTE: What is the first element in the folder name?
+        _, task_id, dir_id = os.path.basename(subject_task_folder).split("_")
+        task_ent = f"task-{task_id}"
+        dir_ent = f"dir-{dir_id}"
+        # TODO: Rename variable
+        filenamex = os.path.basename(subject_task_folder)
```

Comment:
Did you figure out what the first element was in the folder name? I'd like to replace `filenamex`.

Reply:
I think it's just subject I’d

Reply:
In that case, maybe just rename `filenamex` to `run_foldername`.